crates/api_common/src/request.rs

```diff
@@ -63,47 +64,45 @@ pub async fn fetch_link_metadata(url: &Url, context: &LemmyContext) -> LemmyResu
     .await?
     .error_for_status()?;
 
-  // In some cases servers send a wrong mime type for images, which prevents thumbnail
-  // generation. To avoid this we also try to guess the mime type from file extension.
-  let content_type = mime_guess::from_path(url.path())
-    .first()
-    // If you can guess that its an image type, then return that first.
-    .filter(|guess| guess.type_() == mime::IMAGE)
-    // Otherwise, get the content type from the headers
-    .or(
-      response
-        .headers()
-        .get(CONTENT_TYPE)
-        .and_then(|h| h.to_str().ok())
-        .and_then(|h| h.parse().ok()),
-    );
-
-  let opengraph_data = {
-    // if the content type is not text/html, we don't need to parse it
-    let is_html = content_type
-      .as_ref()
-      .map(|c| {
-        (c.type_() == mime::TEXT && c.subtype() == mime::HTML)
-      ||
-      // application/xhtml+xml is a subset of HTML
-      (c.type_() == mime::APPLICATION && c.subtype() == "xhtml")
-      })
-      .unwrap_or(false);
-    if !is_html {
-      Default::default()
-    } else {
+  let mut content_type: Option<Mime> = response
+    .headers()
+    .get(CONTENT_TYPE)
+    .and_then(|h| h.to_str().ok())
+    .and_then(|h| h.parse().ok());
+
+  let mut opengraph_data = Default::default();
+
+  if let Some(c) = &content_type {
+    // application/xhtml+xml is a subset of HTML
+    let application_xhtml: Mime = "application/xhtml+xml".parse()?;
```

Comment:
You can use `.expect()` with `#[allow(clippy::expect_used)]`

Reply:
Please don't add those though, you should be able to do either `.into()` or `.with_lemmy_type(LemmyErrorType::InvalidUrl)?;`

Reply:
The point of using expect was to make it const, which means it cant return an error. If we leave it as `let` then the current code is fine.

Reply:
I left this as is because you [can't call parse in a const definition](https://doc.rust-lang.org/std/keyword.const.html#compile-time-evaluable-functions).